Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -277,7 +277,7 @@ private static void ForceEnumeration<T>(T subject, Func<T, IEnumerable> enumerab
     /// current <see cref="object"/>.
     /// </summary>
     [Pure]
-    public static ObjectAssertions Should(this object actualValue)
+    public static ObjectAssertions Should([System.Diagnostics.CodeAnalysis.NotNull] this object actualValue)
```

Comment:
Why not add the attribute to _all_ `Should()` methods?

Reply:
That's a very good question! I think we should add the [NotNull] attribute on all `Should` methods that return a `ReferenceTypeAssertions<,>`. So I started to write a test and noticed that I could not assert `parameterInfo.IsDecoratedWith<NotNullAttribute>`, hence #2385. 😄

Reply:
> So I started to write a test and noticed that I could not assert parameterInfo.IsDecoratedWith<NotNullAttribute>, hence https://github.com/fluentassertions/fluentassertions/pull/2385. 😄

I don't think we need to have tests for that.

Reply:
Well, at least it was very interesting to dive into all the assertion types, I learnt a lot. And the test is now written anyway, alongside other tests on the `Should` methods. But this pull request could also be merged without the tests added in `AssertionExtensionsSpecs` and thus without requiring #2385 to also be merged.